### PR TITLE
prettier precommit using husky / lint-staged

### PR DESCRIPTION
**Pull request by bastianwegge**
This PR is enabling prettier to reformat all committed code files.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,16 +12,22 @@
...
     "migrate": "knex migrate:latest",
     "rollback": "knex migrate:rollback",
     "watch": "spin watch",
     "heroku-postbuild": "rm -rf build && npm run build && npm run seed",
-    "cli": "node tools/cli"
+    "cli": "node tools/cli",
+    "precommit": "lint-staged"
+  },
+  "lint-staged": {
+    "*.{js,json,css}": ["prettier --write", "git add"]
```
Comment (mairh) [inline package.json]: shouldn't you add `jsx` extension too?

Comment (vlasenko) [reply package.json]: And `scss` extension

Comment (mairh) [reply package.json]: since we officially support `less`. That should be added too by default

Comment (bastianwegge) [reply package.json]: There currently is no .less file and I don't see any "we officially support less" sign. You're sure?

**Comment by mairh** [comment]
Are you sure you have run prettier on all files? I see only user module from the client side.

File: package.json (hunk 1 above)
Comment (mairh) [reply package.json]: Well, maybe it's not needed, but we have less support by default using `spinjs`

**Comment by bastianwegge** [comment]
@mairh I'd vote against reformatting the whole codebase on one swoop. At least that was not my intention. I installed a pre-commit hook (as the name of the PR implies) that takes all committed files (of your current commit) and reformats them.

**Comment by mairh** [comment]
That's perfectly fine.

**Comment by vlasenko** [comment]
@bastianwegge Then can we not reformat the files in this PR at all?
I would vote to reformat all the files in one commit right after we merge this PR

**Comment by bastianwegge** [comment]
closing because of master merge that caused all merged files to be reformatted

**Pull request closed without merging**